src/lib/FlightTasks/tasks/FlightTaskManual.cpp

```diff
@@ -80,13 +80,40 @@ bool FlightTaskManual::_evaluateSticks()
 		_sticks_expo(0) = math::expo_deadzone(_sticks(0), _xy_vel_man_expo.get(), _stick_dz.get());
 		_sticks_expo(1) = math::expo_deadzone(_sticks(1), _xy_vel_man_expo.get(), _stick_dz.get());
 		_sticks_expo(2) = math::expo_deadzone(_sticks(2), _z_vel_man_expo.get(), _stick_dz.get());
+		_sticks_expo(3) = math::expo_deadzone(_sticks(3), _yaw_expo.get(), _stick_dz.get());
+
+		// Only switch the landing gear up if the user switched from gear down to gear up.
+		// If the user had the switch in the gear up position and took off ignore it
+		// until he toggles the switch to avoid retracting the gear immediately on takeoff.
+		int8_t gear_switch = _sub_manual_control_setpoint->get().gear_switch;
+
+		if (!_constraints.landing_gear) {
```

Comment:
I think `_evaluateSticks()` should only get the gear switch state and all the logic executing on it should be in a different function, here `_applyGearSwitch()`